### PR TITLE
[Enhance] Update docstring of cross entropy loss

### PR DESCRIPTION
**Pull request by chhluo**
## Motivation

This is part of Mask2Former( PR #6938 )

## Modification

1. update docstring in `binary_cross_entropy`. When `pred`  have shape (N, 1), it will be  expanded to onehot, while for `pred` with shape (N, ), it will not be expanded to onehot.

### CODE REVIEW

File: mmdet/models/losses/cross_entropy_loss.py [hunk 1]
```diff
@@ -171,6 +181,7 @@ def __init__(self,
                  use_sigmoid=False,
                  use_mask=False,
                  reduction='mean',
+                 bce_expand_one_hot=True,
```
Comment (AronLin) [inline mmdet/models/losses/cross_entropy_loss.py:184]: I have a question.  Wich is better to put this arg in `_init_` or in `forward`.
Can @BIGWangYuDong @jbwang1997 give some advice?
If put in `_init_`, it is convenient to switch the loss in the config file directly, but the disadvantage is that ce loss and mask ce loss do not have this parameter. I don't know if it is suitable or not to put an arg that only belongs to a part of `CrossEntropyLoss` in `_init_`.

Comment (jbwang1997) [reply mmdet/models/losses/cross_entropy_loss.py:184]: The original bce loss can directly input onehot labels. When the `pred` and `label` have the same `dim`, the program will skip the `_expand_onehot_labels` function. So I think it's unnecessary to set `bce_expand_one_hot`.

Comment (AronLin) [reply mmdet/models/losses/cross_entropy_loss.py:184]: > The original bce loss can directly input onehot labels. When the `pred` and `label` have the same `dim`, the program will skip the `_expand_onehot_labels` function. So I think it's unnecessary to set `bce_expand_one_hot`.

It seems that the labels are not one-hot in maskformer.

Comment (jbwang1997) [reply mmdet/models/losses/cross_entropy_loss.py:184]: Yes, the label can also be float if it doesn't pass the `bce_expand_one_hot` function.

Comment (chhluo) [reply mmdet/models/losses/cross_entropy_loss.py]: yes, `bce_expand_one_hot` is unnecessary. The docstring is misleading. I have updated it.

**Review by jbwang1997 — APPROVED** [review]
LGTM

File: mmdet/models/losses/cross_entropy_loss.py [hunk 2]
```diff
@@ -81,7 +81,10 @@ def binary_cross_entropy(pred,
     """Calculate the binary CrossEntropy loss.
 
     Args:
-        pred (torch.Tensor): The prediction with shape (N, 1).
+        pred (torch.Tensor): The prediction with shape (N, 1) or (N, ).
+            When pred is of shape (N, 1) it will be expanded to one-hot
```
Comment (ZwwWayne) [inline mmdet/models/losses/cross_entropy_loss.py:85]: what does `it will be expanded to one-hot format` mean? Does this mean pred's shape will be changed from what to what?
